sdk/python/kfp/dsl/types/artifact_types.py

```diff
@@ -90,13 +91,18 @@ def _get_path(self) -> Optional[str]:
         return None
 
     def _set_path(self, path: str) -> None:
-        if path.startswith(_GCS_LOCAL_MOUNT_PREFIX):
-            path = 'gs://' + path[len(_GCS_LOCAL_MOUNT_PREFIX):]
-        elif path.startswith(_MINIO_LOCAL_MOUNT_PREFIX):
-            path = 'minio://' + path[len(_MINIO_LOCAL_MOUNT_PREFIX):]
-        elif path.startswith(_S3_LOCAL_MOUNT_PREFIX):
-            path = 's3://' + path[len(_S3_LOCAL_MOUNT_PREFIX):]
-        self.uri = path
+        self.uri = convert_local_path_to_remote_path(path)
+
+
+def convert_local_path_to_remote_path(path: str) -> str:
+    if path.startswith(_GCS_LOCAL_MOUNT_PREFIX):
+        return 'gs://' + path[len(_GCS_LOCAL_MOUNT_PREFIX):]
+    elif path.startswith(_MINIO_LOCAL_MOUNT_PREFIX):
+        return 'minio://' + path[len(_MINIO_LOCAL_MOUNT_PREFIX):]
+    elif path.startswith(_S3_LOCAL_MOUNT_PREFIX):
+        return 's3://' + path[len(_S3_LOCAL_MOUNT_PREFIX):]
+    else:
+        raise ValueError(f'Unknown remote path for local path: {path}')
```

Comment:
In respond to your question from the other thread, is this behavior change necessary? 
Old behavior was local path would be preserved as-is, but new behavior throws an error.

Reply:
Good catch. This is not necessary. I removed and added tests for the correct behavior. Continuing the discussion on the other thread.